tests/rust_exec_tests/src/lib.rs

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! musl_fsbase_fix
+
+#[macro_export]
+macro_rules! musl_fsbase_fix {
+    () => {
+        /// Set FSBASE
+        ///
+        /// Overwrite the only location in musl, which uses the `arch_prctl` syscall
+        #[no_mangle]
+        #[inline(never)]
+        pub extern "C" fn __set_thread_area(p: *mut std::ffi::c_void) -> std::ffi::c_int {
```

Comment:
couldn't you theoretically make a trivial crate to encapsulate this and then just link it with cargo to both shims?

Reply:
> couldn't you theoretically make a trivial crate to encapsulate this and then just link it with cargo to both shims?

my tests doing exactly this didn't succeed